src/site/content/en/web-dev-basics-one/fetch-api-error-handling/index.md

```diff
@@ -0,0 +1,234 @@
+1. Provide helpful messaging that explains what went wrong, and what options the user can take.
+1. Based on the available options, present a "Try again" button to the user.
+1. Behind the scenes, send the details of the error to your error-tracking service, or to the back-end. This action logs the error so it can be diagnosed at a later stage.
+
+```js
+try {
+  const response = await fetch('https://website');  
+} catch (error) {
+  // TypeError: Failed to fetch
+  console.log('There was an error', error);
+}
+```
+
+At a later stage, while you diagnose the error that you logged, you can write a [test case](https://en.wikipedia.org/wiki/Test_case) to catch such an error before your users are aware something is wrong. Depending on the error, the test could be a unit, integration, or acceptance test.
+
+### When the network status code represents an error
+
+This code example makes a request to an HTTP testing service that always responds with the HTTP status code [429 Too Many Requests](https://developer.mozilla.org/docs/Web/HTTP/Status/429). Interestingly, the response does not reach the `catch` block. A 404 status, amongst certain other status codes, does return [a network error](https://developer.mozilla.org/docs/Web/API/Fetch_API/Using_Fetch#checking_that_the_fetch_was_successful) but instead resolves normally.
+
+To check that the HTTP status code was successful, you can use any of the following options:
+
+-  Use the [Response.ok](https://developer.mozilla.org/docs/Web/API/Response/ok) property to determine whether the status code was in the range from 200 to 299.
+-  Use the [Response.status](https://developer.mozilla.org/docs/Web/API/Response/status) property to determine whether the response was successful.
+-  Use any other metadata, such as [Response.headers](https://developer.mozilla.org/docs/Web/API/Response/headers), to assess whether the response was successful.
+
+```js
+let response;
+
+try {
+  response = await fetch('https://httpbin.org/status/429');  
+} catch (error) {
+  console.log('There was an error', error);
+}
+
+// Uses the 'optional chaining' operator
+if (response?.ok) {
+  console.log('Use the response here!');
+} else {
+  console.log('HTTP Response Code: ' + response?.status)
+}
+```
+
+The best practice is to work with people in your organization and team to understand potential HTTP response status codes. Backend developers, developer operations, and service engineers can sometimes provide unique insight into possible edge cases that you might not anticipate.
+
+{% Aside %}
+Note from the author: I was recently scrolling through an infinite scrolling list on a website, trying to quickly reach the bottom. Eventually, the page seemed to break with no helpful messaging. Upon inspecting the network request via DevTools, it became clear that the HTTP status code error `429 Too Many Requests` was returned. In an ideal scenario, such internal errors are surfaced to the web page interface.
```

Comment:
Our guidance advises against the first person perspective ("I" or "we"), instead preferring to use the second person (i.e., "you"). If you could rework this to be more general rather than personal, that would align better with our style guide.

Reply:
I'm ok with this, it turns it from a vague "this could happen" to "this actually happened to me", and I think means we don't need to go into too much detail about solutions.